### PR TITLE
Fix typo about semaphore

### PR DESCRIPTION
**Pull request by junmin-kim**
Replace `seamahpore` with `semaphore`

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [d87343ad01fe16909b99404edcb93d175584e7ba] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d87343ad01fe16909b99404edcb93d175584e7ba] - Code Rule Check OK.

**Comment by YiHyunJin** [comment]
Good

File: os/arch/xtensa/src/esp32/esp32_i2c.c [hunk 1]
```diff
@@ -362,7 +362,7 @@ static int i2c_esp32_isr(int irq, FAR void *context, FAR void *arg)
 					break;
 				}
 
-				/* not need to give seamphore, do work in isr itself*/
+				/* not need to give semaphore, do work in isr itself*/
```
Comment (jeongchanKim) [inline os/arch/xtensa/src/esp32/esp32_i2c.c]: Please add space before closing the comment `*/`

Comment (junmin-kim) [reply os/arch/xtensa/src/esp32/esp32_i2c.c]: You're right. I'll fix it

**Comment by seinfra** [comment]
Target : [ce03d7ccb93b92ff3dabaa0710a5dbaf9ac692a9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ce03d7ccb93b92ff3dabaa0710a5dbaf9ac692a9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3d8eceff89e4372e06c9a48e6f2d891b99611a6e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3d8eceff89e4372e06c9a48e6f2d891b99611a6e] - Code Rule Check (C++) OK.

**Pull request merged**